cs/src/Connections/TunnelConnectionOptions.cs

```diff
@@ -0,0 +1,63 @@
+// <copyright file="TunnelConnectionOptions.cs" company="Microsoft">
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// </copyright>
+
+using System.Threading;
+using Microsoft.DevTunnels.Contracts;
+
+namespace Microsoft.DevTunnels.Connections;
+
+/// <summary>
+/// Options for a tunnel host or client connection.
+/// </summary>
+/// <seealso cref="ITunnelHost.ConnectAsync(Tunnel, TunnelConnectionOptions?, CancellationToken)"/>
+public class TunnelConnectionOptions
+{
+    /// <summary>
+    /// Gets or sets a value indicating whether the connection will be automatically retried after
+    /// a connection failure.
+    /// </summary>
+    /// <remarks>
+    /// The default value is true. When enabled, retries continue until the connection is
+    /// successful, the cancellation token is cancelled, or an unrecoverable error is encountered.
+    ///
+    /// Recoverable errors include network connectivity issues, authentication issues (e.g. expired
+    /// access token which may be refreshed before retrying), and service temporarily unavailable
+    /// (HTTP 503). For rate-limiting errors (HTTP 429) only a limited number of retries are
+    /// attempted before giving up.
+    ///
+    /// Retries are performed with exponential backoff, starting with a 100ms delay and doubling
+    /// up to a maximum 12s delay, with further retries using the same max delay.
+    ///
+    /// Note after the initial connection succeeds, the host or client may still become disconnected
+    /// at any time after that. In that case the <see cref="EnableReconnect" /> option controls
```

Comment:
Furthermore, reconnections might be happening about once per week for long-running clients. That's when we do a deployment and Relay pods are restarted.

Reply:
Added comments about that.